kubernetes/resource_kubernetes_mutating_webhook_configuration.go

```diff
@@ -0,0 +1,325 @@
+						},
+						"client_config": {
+							Type:        schema.TypeList,
+							Description: webhookDoc["clientConfig"],
+							Required:    true,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: webhookClientConfigFields(),
+							},
+						},
+						"failure_policy": {
+							Type:        schema.TypeString,
+							Description: webhookDoc["failurePolicy"],
+							Optional:    true,
+							Default:     "Fail",
+						},
+						"match_policy": {
+							Type:        schema.TypeString,
+							Description: webhookDoc["matchPolicy"],
+							Optional:    true,
+							Default:     "Equivalent",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Description: webhookDoc["name"],
+							Required:    true,
+						},
+						"namespace_selector": {
+							Type:        schema.TypeList,
+							Description: webhookDoc["namespaceSelector"],
+							Optional:    true,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: labelSelectorFields(true),
+							},
+						},
+						"object_selector": {
+							Type:        schema.TypeList,
+							Description: webhookDoc["objectSelector"],
+							Optional:    true,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: labelSelectorFields(true),
+							},
+						},
+						"reinvocation_policy": {
+							Type:        schema.TypeString,
+							Description: webhookDoc["reinvocationPolicy"],
+							Optional:    true,
+							Default:     "Never",
+						},
+						"rule": {
+							Type:        schema.TypeList,
+							Description: webhookDoc["rules"],
+							Required:    true,
+							MinItems:    1,
+							Elem: &schema.Resource{
+								Schema: ruleWithOperationsFields(),
+							},
+						},
+						"side_effects": {
+							Type:        schema.TypeString,
+							Description: webhookDoc["sideEffects"],
+							Optional:    true,
+						},
+						"timeout_seconds": {
+							Type:        schema.TypeInt,
+							Description: webhookDoc["timeoutSeconds"],
+							Default:     10,
+							Optional:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceKubernetesMutatingWebhookConfigurationCreate(d *schema.ResourceData, meta interface{}) error {
+	var err error
+
+	conn, err := meta.(KubeClientsets).MainClientset()
+	if err != nil {
+		return err
+	}
+
+	cfg := admissionregistrationv1.MutatingWebhookConfiguration{
+		ObjectMeta: expandMetadata(d.Get("metadata").([]interface{})),
+		Webhooks:   expandMutatingWebhooks(d.Get("webhook").([]interface{})),
+	}
+
+	log.Printf("[INFO] Creating new MutatingWebhookConfiguration: %#v", cfg)
+
+	res := &admissionregistrationv1.MutatingWebhookConfiguration{}
+
+	usev1beta1, err := useAdmissionregistrationV1beta1(conn)
+	if err != nil {
+		return err
+	}
+	if usev1beta1 {
+		resv1beta1 := &admissionregistrationv1beta1.MutatingWebhookConfiguration{}
+		reqv1beta1 := &admissionregistrationv1beta1.MutatingWebhookConfiguration{}
+		copier.Copy(reqv1beta1, cfg)
```

Comment:
So the logic here (and in the other functions) is:

1. We have determined want to use **v1beta1** of the API
2. On line 130 we created `cfg` using the type from **v1** (because our expander/flattener functions use the types from v1)
3. On line 146 we copy the data from `cfg` into `reqv1beta1` which uses type from **v1beta1** 
4. On line 147 we use `reqv1beta1` as the argument to Create() using `v1beta1` of the API, putting the response in `resv1beta1` which also uses the type from **v1beta1**
5. Finally, on line 148 we copy the data from `resv1beta1` into `res` which uses the type **v1** so it can be handed over to the rest of the logic in the function

Reply:
ohh, I didn't realize copying `cfg` into `reqv1beta1` would keep any aspects of v1beta1. That makes more sense, knowing it's not a 100% replaced or overwritten. Thanks

Reply:
Yeah - both types are actually the same we are just copying one into the other here because we can't convert or cast one type to the other, and it's more convenient than duplicating all the flattener/expander code to support both API versions. 😓